File: mobile/lib/shared/models/product.dart
class Product {
  final String id;
  final String title;
  final String category;
  final double price;
  final double rating;
  final int ratingCount;
  final String imageUrl;
  final String description;

  const Product({
    required this.id,
    required this.title,
    required this.category,
    required this.price,
    required this.rating,
    required this.ratingCount,
    required this.imageUrl,
    required this.description,
  });
}

File: src/features/products/models/Product.ts
export type ProductCategory =
  | "Elektronik"
  | "Tøj"
  | "Bøger"
  | "Hjem"
  | "Skønhed"
  | "Legetøj"
  | "Sport"
  | "Andet";

export type Product = {
  id: string;
  sellerId: string;
  sellerName: string;
  title: string;
  description: string;
  price: number;
  currency: "DKK" | "SYP";
  imageUrl: string;
  category: ProductCategory;
  rating: number; // 0..5
  ratingCount: number;
  prime: boolean;
  createdAt: string; // ISO
};
